lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java

```diff
@@ -225,6 +225,31 @@ public UnsignedTransaction putToContractTxUnsigned(
         });
     }
 
+    /**
+     * Creates a new {@link UnsignedTransaction} which sends the given amount of NanoErgs and a
+     * newly minted token to the given contract.
+     *
+     * @param contract     contract to send the newly minted token to
+     * @param tokenBuilder receives the id of the token to mint, must return the new token
+     * @return unsigned transaxtion
+     */
+    public UnsignedTransaction mintTokenToContractTxUnsigned(ErgoContract contract, Function<String, Eip4Token> tokenBuilder) {
+        if (!tokensToSpend.isEmpty()) {
+            throw new IllegalArgumentException("Mint token not useable with spending tokens");
+        }
+
+        return buildTxWithDefaultInputs(txB -> {
+            OutBox newBox = txB.outBoxBuilder()
+                .value(amountToSpend)
+                .contract(contract)
+                .mintToken(tokenBuilder.apply(txB.getInputBoxes().get(0).getId().toString()))
```

Comment:
This `txB.getInputBoxes().get(0).getId().toString()` can be encapsulated in a method. It is found in many places

Reply:
Can you give more information? This is the only place I've found that phrase in appkit.

Reply:
search `get(0).getId().toString()` part

Reply:
Two matches
* this one
* TxBuilderSpec (which tests if this code worked correctly)